tools/slicec-cs/src/encoding.rs

```diff
@@ -78,6 +78,8 @@ fn encode_type(
         concrete_typeref => {
             let value = if type_ref.is_optional && type_ref.is_value_type() {
                 format!("{param}.Value")
+            } else if type_ref.is_optional && !type_ref.is_reference_type() {
```

Comment:
Maybe add a comment about how this handles custom types?
Otherwise this branch seems redundant on the face of it.

Reply:
Can we just check for custom type? I think would make the intent clear

```suggestion
            } else if type_ref.is_optional && matches!(type_ref.concrete_type(), Types::CustomType(_)) {
```

Reply:
I removed is_reference_type as suggested by Jose.